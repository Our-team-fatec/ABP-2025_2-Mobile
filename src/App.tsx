import { NavigationContainer } from "@react-navigation/native";
import { createNativeStackNavigator } from "@react-navigation/native-stack";
import Login from "./screens/login/login";
import Cadastro from "./screens/cadastro/cadastro";

export type RootStackParamList = {
  Login: undefined;
  Cadastro: undefined;
};

const Stack = createNativeStackNavigator<RootStackParamList>();

export default function App() {
  return (
    <NavigationContainer>
      <Stack.Navigator initialRouteName="Login" screenOptions={{ headerShown: false }}>
        <Stack.Screen name="Login" component={Login} />
        <Stack.Screen name="Cadastro" component={Cadastro} />
      </Stack.Navigator>
    </NavigationContainer>
  );
}
