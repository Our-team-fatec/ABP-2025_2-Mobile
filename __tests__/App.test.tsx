import React from 'react';
import renderer, { act } from 'react-test-renderer';
import App from '../src/App';

describe('App Component', () => {
  it('deve renderizar corretamente', async () => {
    let tree;
    await act(async () => {
      tree = renderer.create(<App />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  }, 20000);
});