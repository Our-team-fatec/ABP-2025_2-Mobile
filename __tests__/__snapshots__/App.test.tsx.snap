// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component deve renderizar corretamente 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
      ]
    }
  />
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      activityState={2}
      aria-hidden={false}
      collapsable={false}
      fullScreenSwipeShadowEnabled={true}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      hasLargeHeader={false}
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onHeaderHeightChange={[Function]}
      onNativeDismissCancelled={[Function]}
      onSheetDetentChanged={[Function]}
      onTransitionProgress={[Function]}
      onWillAppear={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      screenId="Login-bTCRAhpTPeNRQytuHW6jM"
      sheetAllowedDetents={
        [
          1,
        ]
      }
      sheetCornerRadius={-1}
      sheetElevation={24}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetInitialDetent={0}
      sheetInitialDetentIndex={0}
      sheetLargestUndimmedDetent={-1}
      sheetLargestUndimmedDetentIndex={-1}
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": undefined,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenContentWrapper
        collapsable={false}
        style={
          [
            {
              "flex": 1,
            },
            [
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ],
          ]
        }
      >
        <RCTSafeAreaView
          style={
            {
              "backgroundColor": "#fafcfa",
              "flex": 1,
            }
          }
        >
          <View
            onLayout={[Function]}
            style={
              [
                {
                  "flex": 1,
                },
                {
                  "paddingBottom": 0,
                },
              ]
            }
          >
            <RCTScrollView
              contentContainerStyle={
                {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "justifyContent": "center",
                  "paddingHorizontal": 20,
                  "paddingVertical": 32,
                }
              }
              keyboardShouldPersistTaps="handled"
            >
              <View>
                <View
                  style={
                    {
                      "maxWidth": 420,
                      "width": "100%",
                    }
                  }
                >
                  <View
                    style={
                      {
                        "alignItems": "center",
                        "marginBottom": 30,
                      }
                    }
                  >
                    <Image
                      source={
                        {
                          "testUri": "../../../assets/favicon.png",
                        }
                      }
                      style={
                        {
                          "borderRadius": 20,
                          "height": 80,
                          "marginBottom": 15,
                          "width": 80,
                        }
                      }
                    />
                    <Text
                      style={
                        {
                          "fontFamily": "Arial",
                          "fontSize": 24,
                          "fontWeight": "bold",
                          "marginBottom": 5,
                        }
                      }
                    >
                      DaVinciPets
                    </Text>
                    <Text
                      style={
                        {
                          "color": "#666",
                          "fontFamily": "Arial",
                          "fontSize": 16,
                          "fontWeight": "300",
                          "textAlign": "center",
                        }
                      }
                    >
                      Entre na sua conta
                    </Text>
                  </View>
                  <View
                    style={
                      {
                        "backgroundColor": "#ffffff",
                        "borderRadius": 15,
                        "elevation": 5,
                        "padding": 30,
                        "shadowColor": "#000",
                        "shadowOffset": {
                          "height": 2,
                          "width": 0,
                        },
                        "shadowOpacity": 0.1,
                        "shadowRadius": 8,
                        "width": "100%",
                      }
                    }
                  >
                    <Text
                      style={
                        {
                          "fontFamily": "Arial",
                          "fontSize": 16,
                          "textAlign": "center",
                        }
                      }
                    >
                      Fazer Login
                    </Text>
                    <View
                      style={
                        {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "marginBottom": 8,
                        }
                      }
                    >
                      <Text
                        color="#666"
                        name="email"
                        size={16}
                        testID="MaterialIcons-mock"
                      >
                        MaterialIcons
                      </Text>
                      <Text
                        style={
                          {
                            "color": "#333",
                            "fontFamily": "Arial",
                            "fontSize": 14,
                            "marginLeft": 6,
                          }
                        }
                      >
                        E-mail
                      </Text>
                    </View>
                    <TextInput
                      autoCapitalize="none"
                      keyboardType="email-address"
                      onBlur={[Function]}
                      onChangeText={[Function]}
                      placeholder="Digite seu e-mail"
                      selectionColor="#89b490"
                      style={
                        [
                          {
                            "backgroundColor": "#f9f9f9",
                            "borderColor": "#9fb9a4ff",
                            "borderRadius": 8,
                            "borderWidth": 1,
                            "fontSize": 16,
                            "marginBottom": 15,
                            "padding": 12,
                          },
                          undefined,
                        ]
                      }
                    />
                    <View
                      style={
                        {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "marginBottom": 8,
                        }
                      }
                    >
                      <Text
                        color="#666"
                        name="lock"
                        size={16}
                        testID="MaterialIcons-mock"
                      >
                        MaterialIcons
                      </Text>
                      <Text
                        style={
                          {
                            "color": "#333",
                            "fontFamily": "Arial",
                            "fontSize": 14,
                            "marginLeft": 6,
                          }
                        }
                      >
                        Senha
                      </Text>
                    </View>
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "backgroundColor": "#f9f9f9",
                            "borderColor": "#9fb9a4ff",
                            "borderRadius": 8,
                            "borderWidth": 1,
                            "flexDirection": "row",
                            "marginBottom": 15,
                          },
                          undefined,
                        ]
                      }
                    >
                      <TextInput
                        onBlur={[Function]}
                        onChangeText={[Function]}
                        placeholder="Digite sua senha"
                        secureTextEntry={true}
                        selectionColor="#89b490"
                        style={
                          {
                            "flex": 1,
                            "fontSize": 16,
                            "padding": 12,
                          }
                        }
                      />
                      <View
                        accessibilityState={
                          {
                            "busy": undefined,
                            "checked": undefined,
                            "disabled": undefined,
                            "expanded": undefined,
                            "selected": undefined,
                          }
                        }
                        accessibilityValue={
                          {
                            "max": undefined,
                            "min": undefined,
                            "now": undefined,
                            "text": undefined,
                          }
                        }
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          {
                            "opacity": 1,
                            "padding": 12,
                          }
                        }
                      >
                        <Text
                          color="#666"
                          name="visibility-off"
                          size={24}
                          testID="MaterialIcons-mock"
                        >
                          MaterialIcons
                        </Text>
                      </View>
                    </View>
                    <View
                      accessibilityState={
                        {
                          "busy": undefined,
                          "checked": undefined,
                          "disabled": undefined,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={false}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "alignItems": "flex-end",
                          "marginBottom": 15,
                          "opacity": 1,
                        }
                      }
                    >
                      <Text
                        style={
                          {
                            "color": "#89b490",
                            "fontFamily": "Arial",
                            "fontSize": 16,
                            "fontWeight": "500",
                            "textDecorationLine": "underline",
                          }
                        }
                      >
                        Esqueci a senha?
                      </Text>
                    </View>
                    <View
                      accessibilityState={
                        {
                          "busy": undefined,
                          "checked": undefined,
                          "disabled": false,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "alignItems": "center",
                          "backgroundColor": "#89b490",
                          "borderRadius": 8,
                          "marginTop": 10,
                          "opacity": 1,
                          "padding": 15,
                        }
                      }
                    >
                      <Text
                        style={
                          {
                            "color": "#ffffff",
                            "fontSize": 16,
                            "fontWeight": "600",
                          }
                        }
                      >
                        Entrar
                      </Text>
                    </View>
                    <View
                      style={
                        {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "marginVertical": 20,
                        }
                      }
                    >
                      <View
                        style={
                          {
                            "backgroundColor": "#ddd",
                            "flex": 1,
                            "height": 1,
                          }
                        }
                      />
                    </View>
                    <View
                      style={
                        {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "justifyContent": "center",
                          "marginTop": 10,
                        }
                      }
                    >
                      <Text
                        style={
                          {
                            "fontFamily": "Arial",
                            "fontSize": 16,
                            "textAlign": "center",
                          }
                        }
                      >
                        Não tenho uma conta? 
                      </Text>
                      <View
                        accessibilityState={
                          {
                            "busy": undefined,
                            "checked": undefined,
                            "disabled": undefined,
                            "expanded": undefined,
                            "selected": undefined,
                          }
                        }
                        accessibilityValue={
                          {
                            "max": undefined,
                            "min": undefined,
                            "now": undefined,
                            "text": undefined,
                          }
                        }
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          {
                            "opacity": 1,
                          }
                        }
                      >
                        <Text
                          style={
                            {
                              "color": "#89b490",
                              "fontFamily": "Arial",
                              "fontSize": 16,
                              "fontWeight": "500",
                              "textDecorationLine": "underline",
                            }
                          }
                        >
                          Criar conta
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </RCTScrollView>
          </View>
        </RCTSafeAreaView>
      </RNSScreenContentWrapper>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleFontFamily="System"
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        largeTitleFontFamily="System"
        largeTitleFontWeight="700"
        largeTitleHideShadow={false}
        pointerEvents="box-none"
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "position": "absolute",
            "width": "100%",
          }
        }
        title="Login"
        titleColor="rgb(28, 28, 30)"
        titleFontFamily="System"
        titleFontWeight="600"
        topInsetEnabled={true}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
